main.go

```diff
@@ -94,51 +97,99 @@ func mainError() error {
 		}
 	}
 
-	azureConfigValidatorConfig := azureupdate.AzureConfigValidatorConfig{
-		K8sClient: k8sClient,
-		Logger:    newLogger,
-	}
-	azureConfigValidator, err := azureupdate.NewAzureConfigValidator(azureConfigValidatorConfig)
-	if err != nil {
-		return microerror.Mask(err)
+	var azureConfigValidator *azureupdate.AzureConfigValidator
```

Comment:
I'm totally ok with this if you want to have separate validators for each CRD, but for this specific upgrade use case on releases, you should be able to write a single generic validator for all CRs. Instead of deserializing into specific CR, maybe you could deserialize into [ObjectMeta](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)? From there you can do `m.GetLabels()` and have a unified implementation.

There might be cases in the future where having distinct validators for specific types helps with smaller PRs, but I leave the judgment to you. Either way works I guess.

Reply:
I like the idea of abstracting the specific CR when only the ObjectMeta is needed.
But it's likely that in the near future we'll need to deal with the whole CR for validation purposes, so I guess it's ok to work with the full objects already.